src/nvim/getchar.c

```diff
@@ -2072,16 +2069,13 @@ static int vgetorpeek(const int advance)
                 noremap = REMAP_YES;
               else
                 noremap = REMAP_SKIP;
-              i = ins_typebuf(s, noremap,
+              ins_typebuf(s, noremap,
                   0, TRUE, cmd_silent || save_m_silent);
               if (save_m_expr)
                 xfree(s);
             }
             xfree(save_m_keys);
             xfree(save_m_str);
-            if (i == FAIL) {
```

Comment:
Yes it would.
I made this change because after the changes of the previous commit, you can see that the actions in the case of failure and success are just to `continue`, so there's no point in checking what happened.